fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.delinquency.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResultBuilder;
+import org.apache.fineract.portfolio.delinquency.api.DelinquencyApiConstants;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyBucketData;
+import org.apache.fineract.portfolio.delinquency.data.DelinquencyRangeData;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucket;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucketMappings;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucketMappingsRepository;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyBucketRepository;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRange;
+import org.apache.fineract.portfolio.delinquency.domain.DelinquencyRangeRepository;
+import org.apache.fineract.portfolio.delinquency.exception.DelinquencyRangeNotFoundException;
+import org.apache.fineract.portfolio.delinquency.validator.DelinquencyBucketParseAndValidator;
+import org.apache.fineract.portfolio.delinquency.validator.DelinquencyRangeParseAndValidator;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class DelinquencyWritePlatformServiceImpl implements DelinquencyWritePlatformService {
+
+    private final DelinquencyBucketParseAndValidator dataValidatorBucket;
+    private final DelinquencyRangeParseAndValidator dataValidatorRange;
+
+    private final DelinquencyRangeRepository repositoryRange;
+    private final DelinquencyBucketRepository repositoryBucket;
+    private final DelinquencyBucketMappingsRepository repositoryBucketMappings;
+
+    @Override
+    public CommandProcessingResult createDelinquencyRange(JsonCommand command) {
+        DelinquencyRangeData data = dataValidatorRange.validateAndParseUpdate(command);
+        Map<String, Object> changes = new HashMap<>();
+        DelinquencyRange delinquencyRange = updateOrCreateDelinquencyRange(data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyRange.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult updateDelinquencyRange(Long delinquencyRangeId, JsonCommand command) {
+        DelinquencyRangeData data = dataValidatorRange.validateAndParseUpdate(command);
+        DelinquencyRange delinquencyRange = this.repositoryRange.getReferenceById(delinquencyRangeId);
```

Comment:
@josehernandezfintecheandomx EclipseLink will throw immediately "EntityNotFoundException" if the reference does not exist.